src/components/Notification/InactivityMessage.jsx

```diff
@@ -0,0 +1,144 @@
+// React Imports
+import React, { useState, useEffect } from 'react';
+// Inrupt Library Imports
+import { LogoutButton } from '@inrupt/solid-ui-react';
+// Styling Imports
+import styled from 'styled-components';
+// Material UI Base imports; TODO: Update imports to @mui/material when re-styling
+import Button from '@mui/base/Button';
+// Utility imports
+import { removeKeys } from '../../utils';
+
+/**
+ * Inactivity Notification Component - Component that displays a popup modal
+ * after 3 minutes of inactivity, prompting the user to either logout or
+ * continue their session.
+ *
+ * @memberof Notification
+ * @name InactivityMessage
+ */
+
+const InactivityMessage = () => {
+  const [showPopup, setShowPopup] = useState(false);
+  const [activeUser, setActiveUser] = useState(false);
+
+  // Checks for active user by looking for a loggedIn key in local storage
+  useEffect(() => {
+    const activeCheck = localStorage.getItem('loggedIn');
+    setActiveUser(activeCheck === 'true');
+  }, []);
+
+  // Toggles the popup after three minutes of inactivity
+  useEffect(() => {
+    let timer = null;
+
+    const resetTimeout = () => {
+      if (timer) clearTimeout(timer);
+
+      timer = setTimeout(() => {
+        setShowPopup(true);
+      }, 180000);
+    };
+
+    const handleUserActivity = () => {
+      resetTimeout();
+    };
+
+    resetTimeout();
+
+    const eventTypes = ['mousedown', 'mousemove', 'touchstart'];
+
+    eventTypes.forEach((eventType) => {
+      window.addEventListener(eventType, handleUserActivity);
+    });
+
+    return () => {
+      clearTimeout(timer);
+      eventTypes.forEach((eventType) => {
+        window.removeEventListener(eventType, handleUserActivity);
+      });
+    };
+  }, []);
+
+  // Event handler for logout and removing items from localStorage
+  // Returns user to home page upon successful logout
+  const handleLogout = () => {
+    removeKeys();
+  };
+
+  return (
+    showPopup &&
+    activeUser && (
+      <StyledOverlay>
+        <StyledModal>
+          <StyledContainer>
+            <p>You have been inactive for a few minutes now. Would you like to log out?</p>
+            <ButtonsContainer>
+              <StyledButton onClick={() => setShowPopup(false)}>Continue Session</StyledButton>
+              <LogoutButton>
+                <StyledLogoutButton onClick={handleLogout}>Log Out</StyledLogoutButton>
+              </LogoutButton>
+            </ButtonsContainer>
+          </StyledContainer>
+        </StyledModal>
+      </StyledOverlay>
+    )
+  );
+};
+
+const StyledOverlay = styled('div')({
```

Comment:
Been pointed out before, but will likely transition full from styled-components to MUI to keep styling system consistent in this project.

Reply:
Same with Inbox.jsx, NewMessage.jsx, and style.css. Can be left for a future PR.